core/nonce.go

```diff
@@ -18,6 +19,7 @@ const MaxUsed = 65536
 
 // NonceService generates, cancels, and tracks Nonces.
 type NonceService struct {
+	mu       sync.Mutex
```

Comment:
Okay, so you could probably decompose it. I'm not sure it'll be a huge boon, however.